versal_boot_bin/build_versal_boot_bin.sh

```diff
@@ -0,0 +1,148 @@
+#!/bin/bash
+set -ex
+
+XSA_FILE=$1
+UBOOT_FILE=${2:-download}
+ATF_FILE=${3:-download}
+BUILD_DIR=build_boot_bin
+OUTPUT_DIR=output_boot_bin
+
+usage () {
+	echo "usage: $0 system_top.xsa (download | u-boot.elf | <path-to-u-boot-source>) (download | bl31.elf | <path-to-arm-trusted-firmware-source>) [output-archive]"
+	exit 1
+}
+
+depends () {
+	echo "Xilinx $1 must be installed and in your PATH"
+	echo "try: source /opt/Xilinx/Vivado/202x.x/settings64.sh"
+	exit 1
+}
+
+build_u_boot () {
+	export DEVICE_TREE="versal-$board-rev1.1"
+	make distclean
+	make xilinx_versal_virt_defconfig
+	make
+}
+
+build_bl31 () {
+	make distclean
+	git checkout $atf_version
+	make PLAT=versal RESET_TO_BL31=1
+}
+
+### Check command line parameters
+echo $XSA_FILE | grep -q ".xsa" || usage
+
+if [ ! -f $XSA_FILE ]; then
+	echo $XSA_FILE: File not found!
+	usage
+fi
+
+### Check for required Xilinx tools (starting with 2019.2 there is no hsi anymore)
+command -v bootgen >/dev/null 2>&1 || depends bootgen
+
+rm -Rf $BUILD_DIR $OUTPUT_DIR
+mkdir -p $OUTPUT_DIR
+mkdir -p $BUILD_DIR
+unzip $XSA_FILE -d $BUILD_DIR
+board_id=$(grep -E -o 'BoardId="[[:alnum:]]+' $BUILD_DIR/xsa.xml)
+board=${board_id#BoardId=\"}
+cp $BUILD_DIR/system_top.pdi $OUTPUT_DIR/system_top.pdi
+export CROSS_COMPILE=aarch64-linux-gnu-
+
+### Check if UBOOT_FILE is .elf or path to u-boot repository
+if [ "$UBOOT_FILE" != "" ] && [ -d $UBOOT_FILE ]; then
+### Build u-boot.elf
+(
+	cd $UBOOT_FILE
+	build_u_boot
+)
+	cp $UBOOT_FILE/u-boot.elf $OUTPUT_DIR/u-boot.elf
+elif [ "$UBOOT_FILE" == "download" ]; then
+(
+	cd $BUILD_DIR
+	git clone https://github.com/Xilinx/u-boot-xlnx
+	cd u-boot-xlnx
+	build_u_boot
+	cp u-boot.elf ../../$OUTPUT_DIR/u-boot.elf
```

Comment:
Just make all of the copy commands consistent - see #61, #68, #76 etc (all inside or all outside parenthesis)
I think that on line #68 is  needed one single level up (../) not two (../../). Please check.